packages/core/src/app/ui/autocomplete/Autocomplete.tsx

```diff
@@ -15,7 +15,7 @@ export interface AutocompleteProps {
     inputProps?: any;
     listTestId?: string;
     onToggleOpen?(state: { inputValue: string; isOpen: boolean }): void;
-    onSelect?(item: AutocompleteItem): void;
+    onSelect?(item: AutocompleteItem | null): void;
```

Comment:
Just wondering why we need to add `null` here?

Reply:
Because `Downshift` allows the selected item to be `null`, so I followed that in our `AutocompleteProps`.
```
node_modules/downshift/typings/index.d.ts
```
<img width="500" alt="image" src="https://github.com/bigcommerce/checkout-js/assets/88361607/2d047d6f-a551-4294-899a-3d7dca871aea">

Reply:
If we don't add `null`, does that throw a type error? Asking this because we should try and be strict if we can.

Reply:
Yes, it will throw a TS error.
I added `null` to pass the linter.
```
TS2769: No overload matches this call. 
Overload 1 of 2, '(props: DownshiftProps<AutocompleteItem> | Readonly<DownshiftProps<AutocompleteItem>>): Downshift<...>', gave the following error.     Type '((item: AutocompleteItem) => void) | undefined' is not assignable to type '((selectedItem: AutocompleteItem | null, stateAndHelpers: ControllerStateAndHelpers<AutocompleteItem>) => void) | undefined'.       Type '(item: AutocompleteItem) => void' is not assignable to type '(selectedItem: AutocompleteItem | null, stateAndHelpers: ControllerStateAndHelpers<AutocompleteItem>) => void'.         Types of parameters 'item' and 'selectedItem' are incompatible.           Type 'AutocompleteItem | null' is not assignable to type 'AutocompleteItem'.             Type 'null' is not assignable to type 'AutocompleteItem'.
Overload 2 of 2, '(props: DownshiftProps<AutocompleteItem>, context: any): Downshift<AutocompleteItem>', gave the following error.     Type '((item: AutocompleteItem) => void) | undefined' is not assignable to type '((selectedItem: AutocompleteItem | null, stateAndHelpers: ControllerStateAndHelpers<AutocompleteItem>) => void) | undefined'.
```

Reply:
The above error will show up at `onChange`.

<img width="300" alt="image" src="https://github.com/bigcommerce/checkout-js/assets/88361607/be7335b2-d53e-4aad-b452-4ce6e00805c2">